xbmc/addons/kodi-addon-dev-kit/include/kodi/api2/.internal/AddonLib_internal.hpp

```diff
@@ -0,0 +1,112 @@
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+//                 _______    _______                              _______    _
+//  ||        ||  ||     ||  ||     ||  ||     ||  ||  ||     ||  ||     ||  | |
+//  ||        ||  ||     ||  ||     ||  ||\    ||  ||  ||\    ||  ||     ||  | |
+//  ||   /\   ||  ||     ||  ||     ||  ||\\   ||  ||  ||\\   ||  ||         | |
+//  ||  //\\  ||  ||=====||  ||=====    || \\  ||  ||  || \\  ||  ||  ____   | |
+//  || //  \\ ||  ||     ||  ||  \\     ||  \\ ||  ||  ||  \\ ||  ||     ||  |_|
+//  ||//    \\||  ||     ||  ||   \\    ||   \\||  ||  ||   \\||  ||     ||   _
+//  ||/      \||  ||     ||  ||    \\   ||    \||  ||  ||    \||  ||_____||  |_|
+//
+//  ____________________________________________________________________________
+// |____________________________________________________________________________|
+//
+// Note this in order to keep compatibility between the API levels!
+//
+// - Do not use enum's as values, use integer instead and define values as
+//   "typedef enum".
+// - Do not use CPP style to pass structures, define always as pointer with "*"!
+// - Prevent use of structure definition in structure!
+// - Do not include dev-kit headers on Kodi's side where function becomes done,
+//   predefine them, see e.g. xbmc/addons/binary/interfaces/api2/Addon/Addon_File.h.
+//   This is needed to prevent conflicts if Level 2 functions are needed in Level3!
+// - Do not use "#defines" for interface values, use typedef!
+// - Do not use for new parts headers from Kodi, create a translation system to
+//   change add.on side to Kodi's!
+//
+//  ____________________________________________________________________________
+// |____________________________________________________________________________|
+//
+
+#include "../definitions.hpp"
+
+#ifdef BUILD_KODI_ADDON
+  #include "kodi/AEChannelData.h"
+#else
+  #include "cores/AudioEngine/Utils/AEChannelData.h"
+#endif
+
+#include "../../kodi_audioengine_types.h"
+#include "../../xbmc_epg_types.h"
+#include "../../xbmc_pvr_types.h"
+#include "../../kodi_adsp_types.h"
+
+#include <map>
+
+#ifdef _WIN32                   // windows
+#ifndef _SSIZE_T_DEFINED
+typedef intptr_t      ssize_t;
+#define _SSIZE_T_DEFINED
+#endif // !_SSIZE_T_DEFINED
+#endif
+
+/*
+ * This file includes not for add-on developer used parts, but required for the
+ * interface to Kodi over the library.
+ */
+
+extern "C"
+{
+
+struct DemuxPacket;
```

Comment:
Ah, has forgot to remove them during reduce of the file for this request, was before needed by related functions. Becomes removed now.